lib/output.js

```diff
@@ -1474,10 +1474,20 @@ function OutputStream(options) {
             output.space();
         }
         if (self.extends) {
+            var parens = self.extends instanceof AST_Binary ||
+                self.extends instanceof AST_Conditional;
             output.print("extends");
-            output.space();
+            if (parens) {
+                output.print("(");
+            } else {
+                output.space();
+            }
             self.extends.print(output);
-            output.space();
+            if (parens) {
+                output.print(")");
+            } else {
+                output.space();
+            }
```

Comment:
It's more trouble than it's worth. `extends` is a special case because it's in the middle of a complex construct - a class. Otherwise the logic would have to be spread in several places.